src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala

```diff
@@ -47,10 +47,11 @@ class AssetsRelation(config: RelationConfig, assetPath: Option[String])(val sqlC
     MetricsSource.getOrCreateCounter(config.metricsPrefix, s"assets.created")
 
   override def insert(df: org.apache.spark.sql.DataFrame, overwrite: scala.Boolean): scala.Unit =
-    df.foreachPartition(rows => {
+    df.foreachPartition((rows: Iterator[Row]) => {
```

Comment:
Scala 2.12 needs some help to get the right overloaded method of `foreachPartition`.